src/common/xo-week-charts/index.js

```diff
@@ -0,0 +1,414 @@
+
+    // 2. Draw horizon area.
+    this._drawHorizonArea(props.data, props.maxValue)
+
+    // 3. Update x axis.
+    svg
+      .select('.x-axis')
+        .call(this.xAxis)
+        .attr('transform', `translate(0, ${props.chartHeight})`)
+        .selectAll('text')
+          .style(X_AXIS_TEXT_STYLE)
+
+    // 4. Update label.
+    svg
+      .select('.label')
+      .attr('dx', 5)
+      .attr('dy', 20)
+        .text(props.label)
+  }
+
+  _handleMouseMove = () => {
+    this.props.onTooltipChange(
+      d3.mouse(this.refs.chart)[0] - HORIZON_AREA_MARGIN
+    )
+  }
+
+  // Update hover area position and text.
+  _updateTooltip (tooltipX) {
+    const date = this.x.invert(tooltipX)
+    const { data } = this.props
+    const index = this.bisectDate(data, date, 1)
+
+    const d0 = data[index - 1]
+    const d1 = data[index]
+
+    // Outside limits.
+    if (d1 === undefined) {
+      return
+    }
+
+    const elem = date - d0.date > d1.date - date ? d1 : d0
+    const x = this.x(elem.date)
+
+    const { props } = this
+    const hover = this.svg.select('.hover-container')
+
+    hover
+      .select('.hover-line')
+        .attr('x1', x)
+        .attr('x2', x)
+
+    hover
+      .select('.hover-text')
+        .attr('dx', x + 5)
+        .attr('dy', props.chartHeight / 2)
+        .text(props.valueRenderer(elem.value))
+  }
+
+  componentDidMount () {
+    // Horizon area ----------------------------------------
+
+    const svg = this.svg = d3
+      .select(this.refs.chart)
+        .append('svg')
+          .attr('transform', `translate(${HORIZON_AREA_MARGIN}, 0)`)
+
+    svg.append('g').attr('class', 'x-axis').style(X_AXIS_STYLE)
+    svg.append('g').attr('class', 'horizon-area')
+    svg.append('text').attr('class', 'label').style(LABEL_STYLE)
+
+    this.x = d3.time.scale()
+    this.y = d3.scale.linear()
+
+    this.bisectDate = d3.bisector(elem => elem.date).left
+
+    this.xAxis = d3
+      .svg
+      .axis()
+      .scale(this.x)
+      .orient('bottom')
+
+    this.line = d3
+      .svg
+      .line()
+      .x(elem => this.x(elem.date))
+      .y(elem => this.y(elem.value))
+
+    // Tooltip ---------------------------------------------
+
+    svg
+      .append('rect')
+        .attr('class', 'mouse-area')
+        .style(MOUSE_AREA_STYLE)
+        .on('mousemove', this._handleMouseMove)
+
+    const hover = svg
+      .append('g')
+        .attr('class', 'hover-container')
+        .style('pointer-events', 'none')
+
+    hover
+      .append('line')
+        .attr('class', 'hover-line')
+        .attr('y1', 0)
+        .style(HOVER_LINE_STYLE)
+
+    hover
+      .append('text')
+        .attr('class', 'hover-text')
+        .style(HOVER_TEXT_STYLE)
+
+    this._draw()
+  }
+
+  componentWillReceiveProps (nextProps) {
+    const { props } = this
+
+    if (!propsEqual(
+      props,
+      nextProps,
+      [ 'chartHeight', 'chartWidth', 'data', 'maxValue' ]
+    )) {
+      this._draw(nextProps)
```

Comment:
IMHO, it would be more standard react if the `_draw` was done in `render` (and in `componentDidMount()` of course).